src/plots/cartesian/axes.js

```diff
@@ -944,9 +952,61 @@ axes.calcTicks = function calcTicks(ax, opts) {
             axes.prepTicks(mockAx, opts);
         }
 
+        if(mockAx.tickmode === 'full domain') {
+            var nt = mockAx.nticks;  // does mockAx have nitkcs?
+            if(nt === undefined) nt = 0;
+            if(nt === 0) {
+              // pass
+            } else if(nt === 1) {
+                tickVals = [0];
+            } else if(nt === 2) {
+                tickVals = [0, 1];
+            } else {
+                var increment = 1 / (nt - 1); // (nt-2) + 1
+                tickVals.push(0);
+                for(var tickIndex = 0; tickIndex < nt - 2; tickIndex++) {
+                    tickVals.push((tickIndex + 1) * increment);
+                }
+                tickVals.push(1);
+            }
+            if(major) {
+                Lib.nestedProperty(ax, 'tickvals').set(tickVals);
+            } else {
+                Lib.nestedProperty(ax.minor, 'tickvals').set(tickVals);
```

Comment:
We should try to avoid altering real attribute values after the `supplyDefaults` step. But you can set an underscored attribute like `_tickvals` if you like, then you just need to look for that in `arrayTicks` (if you're in one of the domain tickmodes). That should avoid the need to reset it later too, I'd imagine.

Also kudos on figuring out `nestedProperty` - but it's overkill here since we know the attribute names, you can just do `ax._tickvals = tickVals` etc - or if you feel like code golf, `(major ? ax : ax.minor)._tickvals = tickVals` 😎